src/claims.rs

```diff
@@ -175,7 +175,8 @@ impl<T> Claims<T> {
         self.expiration.map_or(
             Err(ValidationError::NoClaim(Claim::Expiration)),
             |expiration| {
-                if (options.clock_fn)() > expiration + options.leeway {
+                let expiration = expiration.checked_add_signed(options.leeway);
+                if expiration.is_some() && (options.clock_fn)() > expiration.unwrap() {
```

Comment:
Again, a bit of bikeshedding, but I feel that using `if let` here would be more idiomatic:

```rust
if let Some(expiration) = expiration_with_leeway { // suggestion above applied
    if (options.clock_fn)() > expiration {
        return Err(ValidationError::Expired);
    }
}
Ok(self)
```